nextflow_aws_batch_trigger.py

```diff
@@ -0,0 +1,314 @@
+import boto3
+import base64
+import argparse
+import datetime
+import time
+import logging
+
+now = datetime.datetime.now()
+
+logging.getLogger().setLevel(logging.DEBUG)
+logging.basicConfig(filename="batch_trigger.log", level=logging.INFO)
+
+
+def create_launch_template(
+    ec2_client, launch_template_name, key_name, s3fs_mount, s3_bucket
+):
+    # Use the describe_launch_templates() method to retrieve a list of all launch templates
+    launch_template_response = ec2_client.describe_launch_templates()
+
+    # Extract the LaunchTemplates list from the response
+    launch_templates = launch_template_response["LaunchTemplates"]
+
+    launch_template_names = [i["LaunchTemplateName"] for i in launch_templates]
+
+    if launch_template_name not in launch_template_names:
+        user_data = (
+            "MIME-Version: 1.0\nContent-Type: multipart/mixed; boundary="
```

Comment:
you can apply formatting to this to make it more clear:

f"""
MIME-Version...
mkdir {s3fs_mount}
"""
etc

Reply:
Applied formatting as suggested.